consensus/hotstuff/verification/combined_signer_v3.go

```diff
@@ -0,0 +1,127 @@
+package verification
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/onflow/flow-go/consensus/hotstuff"
+	"github.com/onflow/flow-go/consensus/hotstuff/model"
+	"github.com/onflow/flow-go/consensus/hotstuff/signature"
+	"github.com/onflow/flow-go/crypto"
+	"github.com/onflow/flow-go/crypto/hash"
+	"github.com/onflow/flow-go/model/encoding"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+)
+
+// CombinedSignerV3 creates votes for the main consensus.
+// When a participant votes for a block, it _always_ provide the staking signature
+// as part of their vote. Furthermore, the participant can _optionally_
+// also provide a random beacon signature. Through their staking signature, a
+// participant always contributes to HotStuff's progress. Participation in the random
+// beacon is optional (but encouraged). This allows nodes that failed the DKG to
+// still contribute only to consensus (as fallback).
+// TODO: to be replaced by CombinedSignerV3 for mature V2 solution.
+// The difference between V2 and V3 is that V2 will sign 2 sigs, whereas
+// V3 only sign 1 sig.
+type CombinedSignerV3 struct {
+	staking        module.Local
+	stakingHasher  hash.Hasher
+	beaconKeyStore module.RandomBeaconKeyStore
+	beaconHasher   hash.Hasher
+	signerID       flow.Identifier
```

Comment:
Same here: I think we could just use `module.Local.NodeID()` instead of explicityly storing the node ID again (?)